CRM/Utils/Hook.php

```diff
@@ -155,15 +155,14 @@ public function invoke(
     if (!is_array($names)) {
       // We were called with the old contract wherein $names is actually an int.
       // Symfony dispatcher requires some kind of name.
-      // TODO: Emit a warning, eg
-      // error_log("Warning: hook_$fnSuffix does not give names for its parameters. It will present odd names to any Symfony event listeners.");
+      Civi::log()->warning("hook_$fnSuffix should be updated to pass an array of parameter names to CRM_Utils_Hook::invoke().", ['civi.tag' => 'deprecated']);
```

Comment:
@colemanw the CRM_Core_Error::deprecatedFunctionWarning( wrapper function might not be perfect - but it's more greppable so I think it's preferred